code/datums/components/_component.dm

```diff
@@ -14,8 +14,9 @@
 		stack_trace("Incompatible [type] assigned to a [parent.type]! args: [json_encode(arguments)]")
 		qdel(src, TRUE, TRUE)
 		CRASH("Incompatible [type] assigned to a [parent.type]! args: [json_encode(arguments)]")
-
-	_JoinParent(parent)
+	
+	if(!QDELETED(src))
```

Comment:
does this even fix anything? Surely parent would be the thing being deleted

Reply:
Yeah. I got a runtime when I tried to put the hotel staff outfit on someone on station level. It runtimes because the hotelloving component's parent gets lost.

Reply:
Please fix the implementation on the hotel staff outfit, don't mess with component internals. there's nothing wrong with them right now

Reply:
opted to just leave this alone for now